builtin/providers/aws/resource_aws_db_instance.go

```diff
@@ -74,6 +74,26 @@ func resourceAwsDbInstance() *schema.Resource {
 				Type:     schema.TypeString,
 				Required: true,
 				ForceNew: true,
+				ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
+					value := v.(string)
+					if !regexp.MustCompile(`^[0-9a-z-]$`).MatchString(value) {
```

Comment:
Nope, this should actually be `^[0-9a-z-]+$` (i.e. quantifier `+` is missing, so it's checking a single character only), thanks for noticing, fix is on the way.